graphql/schema/query.graphql

```diff
@@ -56,6 +56,7 @@ type Query {
     before: Time
   ): ProjectEvents!
   projectSettings(projectIdentifier: String! @requireProjectAccess(permission: SETTINGS, access:VIEW)): ProjectSettings!
+  repo(repoId: String! @requireProjectAccess(permission: TASKS, access: VIEW)): RepoRef!
```

Comment:
okay now that I'm thinking about it, I feel like we don't have any built-in support for the condition "If user has task permission for a project, they should also have task permission for the repo" and we would have to write code to support it. 

it is kind of a weird condition but I think we need to have the permission restriction to block access to restricted projects 😭